src/subject/dto/subject-query.dto.ts

```diff
@@ -0,0 +1,23 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { Transform } from 'class-transformer';
+import { Min, Max, IsOptional } from 'class-validator';
+import { QueryPaginationDto } from 'src/common/dto/query-pagination.dto';
+
+export class SubjectQueryDto extends QueryPaginationDto {
+  @ApiProperty({
+    required: false,
+    description: 'ID do professor',
+  })
+  @IsOptional()
+  teacherId?: number;
+
+  @ApiProperty({
+    required: false,
+    description: 'Status de monitoria',
+  })
+  @IsOptional()
+  @Min(1, { message: 'O campo monitorStatus possui um valor inválido' })
+  @Max(5, { message: 'O campo monitorStatus possui um valor inválido' })
+  @Transform(({ value }) => Number(value))
+  monitorStatus?: number;
```

Comment:
O ideal é fazer a validação com o enum mesmo:

```suggestion
  @IsOptional()
  @IsEnum(MonitorStatus)
  @Transform(({ value }) => Number(value))
  monitorStatus?: MonitorStatus;
```

- importação do IsEnum: `import { IsEnum } from 'class-validator';`
- MonitorStatus é o do arquivo `src/monitor/utils/monitor.enum.ts`

PS.: Verificar se o @Transform ainda é necessário usando o IsEnum

Reply:
Referência: https://stackoverflow.com/questions/62306451/nestjs-isenum-dto-validation-and-swagger

Reply:
Correção feita.
O Transform continua sendo necessário, sem ela a rota apresenta erro 400